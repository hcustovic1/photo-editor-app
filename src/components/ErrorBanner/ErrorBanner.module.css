.errorContainer {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px;
  background-color: var(--color-error-bg);
  color: var(--color-error-text);
  border: 1px solid var(--color-error-border);
  border-radius: 8px;
  margin: 20px;
}

.errorIcon {
  font-size: 1.5rem;
  margin-right: 10px;
}

.errorMessage {
  font-size: 1.2rem;
  font-weight: var(--font-weight-bold);
}