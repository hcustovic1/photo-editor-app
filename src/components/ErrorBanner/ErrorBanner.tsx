import React from 'react';
import styles from './ErrorBanner.module.css';

interface ErrorBannerProps {
  message: string;
}

export const ErrorBanner: React.FC<ErrorBannerProps> = ({ message }) => {
  return (
    <div className={styles.errorContainer} role="alert" aria-live="assertive">
      <div className={styles.errorIcon} aria-hidden="true">
        ⚠️
      </div>
      <p className={styles.errorMessage}>{message}</p>
    </div>
  );
};
