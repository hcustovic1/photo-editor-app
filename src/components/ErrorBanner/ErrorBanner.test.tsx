import { render } from '@testing-library/react';
import { describe, test, expect } from 'vitest';
import { ErrorBanner } from './ErrorBanner';

describe('ErrorBanner Component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(
      <ErrorBanner message="Error! Please check your internet connection" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
