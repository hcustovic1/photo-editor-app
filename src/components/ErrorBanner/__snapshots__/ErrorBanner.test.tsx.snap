// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBanner Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-live="assertive"
    class="_errorContainer_a5014f"
    role="alert"
  >
    <div
      aria-hidden="true"
      class="_errorIcon_a5014f"
    >
      ⚠️
    </div>
    <p
      class="_errorMessage_a5014f"
    >
      Error! Please check your internet connection
    </p>
  </div>
</DocumentFragment>
`;
